### PR TITLE
Add ASP.NET example using the .NET SDK

### PR DESCRIPTION
**Pull request by hculea**
This PR adds the newly created SDK example for C#, as exemplified using aspnet.

### CODE REVIEW

File: application/aspnet/sdk-example/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+Build the ASP.NET docker demo
+```
+docker build . -t aspnet-secrethub-demo
+```
+
+Run the docker demo, passing the newly created service credential and your username as environment variables.
+```
+docker run -e SECRETHUB_CREDENTIAL=$(cat $HOME/.secrethub/credential) -e \
```
Comment (Marton6) [inline application/aspnet/sdk-example/README.md:33]:
```suggestion
docker run -e SECRETHUB_CREDENTIAL=$(cat demo_service.cred) -e \
```

We should use the service credential to demonstrate how to do it.

Make sure you try test the guide again after this change to ensure that it still works.

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                }
+                catch
+                {
+                    return "error in fetching username";
+                }
+            }
+            catch
+            {
+                return "error in client creation";
+            }
```
Comment (Marton6) [inline application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:56]: It's nice to have an elegant error for everything, but why are these `try-catch` blocks nested? If you return in the catch block anyway, I think you can just put them next to each other.

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    }
+                }
+                catch
+                {
+                    return "error in fetching username";
+                }
+            }
+            catch
+            {
+                return "error in client creation";
```
Comment (Marton6) [inline application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:55]: Also we should log the message of the exception we catch, to help the user diagnose any issues they encounter.

**Review by Marton6 — COMMENTED** [review]
Looks good so far!

File: application/aspnet/sdk-example/README.md (hunk 1 above)
Comment (hculea) [reply application/aspnet/sdk-example/README.md:33]: Indeed, it does work, I just miscopied the command, but it is already tested. Nice spot!

File: application/aspnet/sdk-example/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# .NET CORE 3.1 base image
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1
+
+# Create new app
+RUN dotnet new webApp -o example --no-https
+
+# Add modified files
+COPY ./example ./example
```
[the remark this answers is not in the document]
Comment (hculea) [reply application/aspnet/sdk-example/Dockerfile:8]: Django, Kitura, Perfect and Rails are the examples that I can find on master, having the same approach.
I think this was also debated back when we were doing the first batch of examples. 
From what I remember, we settled upon moving the files that need to be changed in the container, such that a user going through the repo will know exactly which files would need to be modified in a newly created, clean project in order to incorporate the functionality we present.
If you still believe that copying the full project is the way to go, I will modify this, but, from what I saw, the examples that approach this alternative are those where the project consists entirely of 2 or 3 files. (see node.js, for example). This is not the case for aspnet.

File: application/aspnet/sdk-example/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<p align="center">
+  <img src="https://secrethub.io/img/integrations/aspnet/github-banner.png?v1" alt="ASP.NET + SecretHub" height="230">
+</p>
+<br/>
+
+In contrast with the CLI example, the SDK example requires less interaction with environment variables, as it is a native integration for C#. Following the below steps will result in an welcome message on http://localhost:5000. If any error shall occur, you will receive an appropriate descriptive message at the same address.
```
[the remark this answers is not in the document]
Comment (hculea) [reply application/aspnet/sdk-example/README.md:6]: A good one though, I think that this wording kind of suggests the superiority of this approach, which is not at all what we want. I'm modifying it now.

File: application/aspnet/sdk-example/Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (Marton6) [reply application/aspnet/sdk-example/Dockerfile:8]: Alright, let's keep like this then. Thanks for informing me.

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs (hunk 3 above)
Comment (hculea) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:34]: Let me know if this is more to your liking :D

File: application/aspnet/sdk-example/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (hculea) [reply application/aspnet/sdk-example/README.md]: Let me know if you like this better.

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs (hunk 2 above)
Comment (hculea) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs]: Let me know if I understood you correctly and the actual version is what you suggested.

File: application/aspnet/sdk-example/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (hculea) [reply application/aspnet/sdk-example/README.md]: Let me know if you like the present wording better.

File: application/aspnet/sdk-example/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<p align="center">
+  <img src="https://secrethub.io/img/integrations/aspnet/github-banner.png?v1" alt="ASP.NET + SecretHub" height="230">
+</p>
+<br/>
+
+This example aims to display the use of the native .NET SecretHub integration.  Following the below steps will result in an welcome message on http://localhost:5000. If any error shall occur, you will receive an appropriate descriptive message at the same address.
```
Comment (Marton6) [inline application/aspnet/sdk-example/README.md:6]:
```suggestion
This example demonstrates the use of the [native SecretHub .NET integration](https://github.com/secrethub/secrethub-dotnet) in an ASP.NET application.
Following the steps below will result in an welcome message on http://localhost:5000.
If any error occurs, you will receive a descriptive error message in the console.
```

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs (hunk 3 above)
Comment (Marton6) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:34]: Yeah, it looks good to me :+1:

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs (hunk 2 above)
Comment (Marton6) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs]: Yeah, you got it! :+1:

File: application/aspnet/sdk-example/README.md (hunk 6 above)
Comment (Marton6) [reply application/aspnet/sdk-example/README.md]: I'm also not sure about calling it "the native SecretHub .NET integration". Maybe calling it "the SecretHub .NET client" would be better...

What do you think, @florisvdg?

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            try
+            {
+                user = System.Environment.GetEnvironmentVariable("SECRETHUB_USERNAME");
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                return "error in fetching username";
```
Comment (Marton6) [inline application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:43]:
```suggestion
                return "error fetching username";

```

I'd drop the "in" from all error messages.

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+            string user, username, password;
+
+            try
+            {
+                client = new SecretHub.Client();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                return "error in client creation";
```
Comment (Marton6) [inline application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:34]:
```suggestion
                return "error creating client";
```

Comment (Marton6) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:34]: I'd drop the "in" from all of the error messages.

Comment (hculea) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs]: Done

File: application/aspnet/sdk-example/README.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+secrethub service init --description demo_service \
+--permission read --file demo_service.cred ${SECRETHUB_USERNAME}/demo
+```
+
+Build the ASP.NET docker demo
+```
+docker build . -t aspnet-secrethub-demo
+```
+
+Run the docker demo, passing the newly created service credential and your username as environment variables.
```
Comment (Marton6) [inline application/aspnet/sdk-example/README.md:33]: We should either end all of these sentences with a `.` or none of them (for consistency).

Comment (hculea) [reply application/aspnet/sdk-example/README.md]: Done! :)

**Review by Marton6 — APPROVED** [review]
Looks great! I also tested it and it worked well.

File: application/aspnet/sdk-example/example/example.csproj [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="SecretHub" Version="0.1.0" />
```
Comment (florisvdg) [inline application/aspnet/sdk-example/example/example.csproj:8]: Let's use the latest version here:
```suggestion
    <PackageReference Include="SecretHub" Version="0.2.0" />
```

File: application/aspnet/sdk-example/README.md (hunk 6 above)
Comment (florisvdg) [reply application/aspnet/sdk-example/README.md]: We called it .NET SDK in the guide.

Comment (hculea) [reply application/aspnet/sdk-example/README.md]: Would you liked it changed from "native .NET SecretHub integration" to ".NET SDK"?

File: application/aspnet/sdk-example/example/Pages/Index.cshtml.cs [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+            try
+            {
+                password = client.Resolve(System.Environment.GetEnvironmentVariable("DEMO_PASSWORD"));
+                return "Hello, " + username + "!";
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                return "error reading demo password";
+            }
```
Comment (florisvdg) [inline application/aspnet/sdk-example/example/Pages/Index.cshtml.cs:54]: Should we use `ResolveEnv` here too? Could turn 3 try-catch into a single one.

Comment (hculea) [reply application/aspnet/sdk-example/example/Pages/Index.cshtml.cs]: Let me know if you like this better.

**Review by florisvdg — APPROVED** [review]

**Pull request merged**